src/data/polynomial/mirror.lean

```diff
@@ -0,0 +1,185 @@
+  rw [rev_at_le (h2.trans (nat.le_add_right _ _))],
+  by_cases h3 : p.nat_trailing_degree ≤ n,
+  { rw [←nat.sub_add_eq_add_sub h2, ←nat.sub_sub_assoc h2 h3, mirror, coeff_mul_X_pow',
+        if_pos h3, coeff_reverse, rev_at_le ((nat.sub_le_self _ _).trans h2)] },
+  rw not_le at h3,
+  rw coeff_eq_zero_of_nat_degree_lt (nat.lt_sub_right_iff_add_lt.mpr (nat.add_lt_add_left h3 _)),
+  exact coeff_eq_zero_of_lt_nat_trailing_degree (by rwa mirror_nat_trailing_degree),
+end
+
+lemma mirror_eval_one : p.mirror.eval 1 = p.eval 1 :=
+begin
+  simp_rw [eval_eq_finset_sum, one_pow, mul_one, mirror_nat_degree],
+  refine finset.sum_bij_ne_zero _ _ _ _ _,
+  { exact λ n hn hp, rev_at (p.nat_degree + p.nat_trailing_degree) n },
+  { intros n hn hp,
+    rw finset.mem_range_succ_iff at *,
+    rw rev_at_le (hn.trans (nat.le_add_right _ _)),
+    rw [nat.sub_le_iff, add_comm, nat.add_sub_cancel, ←mirror_nat_trailing_degree],
+    exact nat_trailing_degree_le_of_ne_zero hp },
+  { exact λ n₁ n₂ hn₁ hp₁ hn₂ hp₂ h, by rw [←@rev_at_invol _ n₁, h, rev_at_invol] },
+  { intros n hn hp,
+    use rev_at (p.nat_degree + p.nat_trailing_degree) n,
+    refine ⟨_, _, rev_at_invol.symm⟩,
+    { rw finset.mem_range_succ_iff at *,
+      rw rev_at_le (hn.trans (nat.le_add_right _ _)),
+      rw [nat.sub_le_iff, add_comm, nat.add_sub_cancel],
+      exact nat_trailing_degree_le_of_ne_zero hp },
+    { change p.mirror.coeff _ ≠ 0,
+      rwa [coeff_mirror, rev_at_invol] } },
+  { exact λ n hn hp, p.coeff_mirror n },
+end
+
+lemma mirror_mirror : p.mirror.mirror = p :=
+polynomial.ext (λ n, by rw [coeff_mirror, coeff_mirror,
+  mirror_nat_degree, mirror_nat_trailing_degree, rev_at_invol])
+
+lemma mirror_eq_zero : p.mirror = 0 ↔ p = 0 :=
+⟨λ h, by rw [←p.mirror_mirror, h, mirror_zero], λ h, by rw [h, mirror_zero]⟩
+
+lemma mirror_trailing_coeff : p.mirror.trailing_coeff = p.leading_coeff :=
+by rw [leading_coeff, trailing_coeff, mirror_nat_trailing_degree, coeff_mirror,
+  rev_at_le (nat.le_add_left _ _), nat.add_sub_cancel]
+
+lemma mirror_leading_coeff : p.mirror.leading_coeff = p.trailing_coeff :=
+by rw [←p.mirror_mirror, mirror_trailing_coeff, p.mirror_mirror]
+
+lemma mirror_mul_of_domain {R : Type*} [integral_domain R] (p q : polynomial R) :
+  (p * q).mirror = p.mirror * q.mirror :=
+begin
+  by_cases hp : p = 0,
+  { rw [hp, zero_mul, mirror_zero, zero_mul] },
+  by_cases hq : q = 0,
+  { rw [hq, mul_zero, mirror_zero, mul_zero] },
+  rw [mirror, mirror, mirror, reverse_mul_of_domain, nat_trailing_degree_mul hp hq, pow_add],
+  ring,
+end
+
+lemma mirror_smul {R : Type*} [integral_domain R] (p : polynomial R) (a : R) :
+  (a • p).mirror = a • p.mirror :=
+by rw [←C_mul', ←C_mul', mirror_mul_of_domain, mirror_C]
+
+lemma mirror_neg {R : Type*} [ring R] (p : polynomial R) : (-p).mirror = -(p.mirror) :=
+by rw [mirror, mirror, reverse_neg, nat_trailing_degree_neg, neg_mul_eq_neg_mul]
+
+lemma mirror_irreducible_test {R : Type*} [integral_domain R] {f : polynomial R}
```

Comment:
`irreducible_of_mirror`? I think that name is unambiguous and closer to the naming convention. (Even though it isn't perfect, but the perfect name would be loooong which isn't perfect either.